spec/install/deploy_spec.rb

```diff
@@ -295,6 +295,106 @@
       expect(out).not_to include("You have deleted from the Gemfile")
     end
 
+    it "allows to replace a host with the same host with credentials when allow_deployment_source_credential_changes is true" do
+      install_gemfile(<<-G, :artifice => "endpoint_strict_basic_authentication", :quiet => true)
```

Comment:
are the `install_gemfile` and `lockfile` the same for all these tests? If so we should wrap these tests in a `describe` block and use a `before` block so we're not repeating ourselves.